docs/customizing.md

```diff
@@ -1,17 +1,184 @@
+additiontal resources within the installation media so that during installation, or
+eventually at boot time, additional binaries such as drivers can be included.
 
-## Docker image
+Another option would be to remaster the Elemental Teal by simply using a docker build.
+Elemental Teal is a regular container image, so it is absolutely possible to create
+a new image using a Dockerfile based on Elemental Teal image.
+
+## Customize installation ISO and installation process
+
+In order to adapt the installation ISO a simple approach is to append extra configuration
+files into the ISO root in the same way a registration yaml configuration file
+is added.
+
+### Additional configuration files
+
+Elemental Teal installation can be customized in three different non-exclisive ways. First, including
+some custom elemental client configuration file, second, by including additional cloud-init files to execute at
+boot time, and finally, by including installation hooks.
+
+#### Custom elemental client configuration file
+
+[Elemental client](https://github.com/rancher/elemental-cli) `install`, `upgrade` and `reset` commands can be configured with a
+custom [configuration file](https://rancher.github.io/elemental-toolkit/docs/customizing/general_configuration/). In order to set a custom configuration file in the installation
+media the MachineRegistration resource associated with this ISO should also include
+the elemental client configuration directory, for that purpose, the `install` field
+supports the `config-dir` field. See [MachineRegistration reference](../machineregistration-reference#configelementalinstall) and the example
+below:
+
+```yaml
+apiVersion: elemental.cattle.io/v1beta1
+kind: MachineRegistration
+metadata:
+  name: my-nodes
+  namespace: fleet-default
+spec:
+  ...
+  config:
+    ...
+    elemental:
+      ...
+      install:
+        ...
+        config-dir: "/run/initramfs/live/elemental.conf.d"
+```
+
+Elemental Teal live ISOs, when booted, the ISO root is mounted at `/run/initramfs/live`.
+So in that case, the ISO should contain the custom elemental client configuration file
+as `/elemental.conf.d/config.yaml`.
+
+#### Adding additional cloud-init files at boot
+
+In order to include additional cloud-init files during the installation they need
+to be added to the installation data into the MachineRegistration resource. More specific
+the `config-urls` field it used for this exact purpose. See [MachineRegistration reference](../machineregistration-reference) page.
+
+`config-urls` is a list of string literals where each item is an http url pointing to a
+cloud-init file or a local path of a cloud init file. Note the local path is evaluated at the
+time of execution by the installation media, hence the local path must exist within
+the installation media, commonly an ISO image.
+
+Elemental Teal live ISOs, when booted, the ISO root is mounted at `/run/initramfs/live`,
```

Comment:
```suggestion
Elemental Teal live ISOs, when booted, have the ISO root mounted at `/run/initramfs/live`,
```

Reply:
Duplicate of line #49 ?

Reply:
Yes, I guess this is a leftover of moving pieces up and down